ml/Host.cc

```diff
@@ -437,6 +490,41 @@ void DetectableHost::detect() {
         updateDetectionChart(getRH(), Dur.count() * 1000);
 
         std::this_thread::sleep_for(Seconds{updateEvery()});
+
+        /*control the size of the database table and shrink them if required*/
+        //DB.shrinkAnomalyRateInfoTable(static_cast<int>(Cfg.MaxAnomalyRateInfoTableSize));
+        /*control the age of the data and remove them if required*/
+        //time_t OldestTime = now_realtime_sec() - (Cfg.MaxAnomalyRateInfoAge * 3600);
+        //DB.removeOldAnomalyRateInfo(OldestTime);
+    }
+}
+
+void DetectableHost::getAnomalyRateInfoCurrentRange(std::vector<std::pair<std::string, double>> &V, time_t After, time_t Before) {
+    for (auto &DP : DimensionsMap) {
```

Comment:
DimensionsMap should be locked whenever we want to iterate it.

Reply:
Done.